src/static/js/chapter.js

```diff
@@ -0,0 +1,189 @@
+        var fig_img = all_fig_imgs[index];
+
+        if (fig_img.getAttribute('data-src')) {
+
+          var iframe = document.createElement('iframe');
+
+          //Set up some default attributes
+          iframe.setAttribute('title', fig_img.getAttribute('alt'));
+          iframe.setAttribute('class', 'fig-iframe');
+          if (fig_img.getAttribute('aria-labelledby')) {
+            iframe.setAttribute('aria-labelledby', fig_img.getAttribute('aria-labelledby'));
+          }
+          if (fig_img.getAttribute('aria-describedby')) {
+            iframe.setAttribute('aria-describedby', fig_img.getAttribute('aria-describedby'));
+          }
+          iframe.setAttribute('width', fig_img.dataset.width || "600");
+          iframe.setAttribute('height', fig_img.dataset.height || '371');
+          iframe.setAttribute('seamless', fig_img.dataset.seamless || '');
+          iframe.setAttribute('frameborder', fig_img.dataset.frameborder || '0');
+          iframe.setAttribute('scrolling', fig_img.dataset.scrolling || 'no');
+          iframe.setAttribute('loading', fig_img.dataset.loading || 'lazy');
+          iframe.setAttribute('src', fig_img.dataset.src);
+
+          //The figure should have a link
+          var parentLink = fig_img.parentNode;
+          if (parentLink.nodeName == "A") {
+
+            //Insert the iframe before the link.
+            parentLink.parentNode.insertBefore(iframe, parentLink);
+
+            //Add the fig-mobile class to hide the img in desktop view
+            parentLink.classList.add("fig-mobile");
+          }
+
+          //Add a test image to check we can actually access Google Sheets
+          //as it's sometimes blocked by corporate proxies and the like
+          //have a fallback function to revert if this is the case
+          var google_sheets_pixel = document.createElement('img');
+          google_sheets_pixel.setAttribute('src', 'https://docs.google.com/favicon.ico');
+          google_sheets_pixel.setAttribute('height', '1');
+          google_sheets_pixel.setAttribute('width', '1');
+          google_sheets_pixel.addEventListener('load', googleSheetsPixelLoaded);
+          google_sheets_pixel.addEventListener('error', googleSheetsPixelNotLoaded);
+          window.document.body.appendChild(google_sheets_pixel);
+
+        }
+      };
+
+    } else {
+      gtag('event', 'interactive-figures', { 'event_category': 'user', 'event_label': 'not-enabled', 'value': 0 });
+    }
+  } catch (err) {
+    console.error('Error' + err);
+    gtag('event', 'interactive-figures', { 'event_category': 'user', 'event_label': 'not-enabled', 'value': 0 });
+  }
+}
+
+function setDiscussionCount() {
+  try {
+    if (window.discussion_url) {
+      fetch(window.discussion_url)
+        .then(function (response) { return response.json(); })
+        .then(function (response) {
+          if (!response) {
+            return;
+          }
+
+          var comments = + response.posts_count - 1;

```

Comment:
~Is the `= +` deliberate?~ Oh, I see it was inherited from the previous code.

Reply:
Yup not mine. @rviscomi not sure if there is a reason for this? Resolving it for this PR anyway.